gpttype_adapter.cpp

```diff
@@ -1095,8 +1096,22 @@ ModelLoadResult gpttype_load_model(const load_model_inputs inputs, FileFormat in
             }
             else
             {
-                float multiplier_rope_base = llamamodel->hparams.rope_freq_base_train/10000.0f;
-                rope_freq_base *= multiplier_rope_base;
+                //float multiplier_rope_base = llamamodel->hparams.rope_freq_base_train/10000.0f;
+                //rope_freq_base *= multiplier_rope_base;
+                
+				//Calculate rope_freq_base using the gradientAI formula, solar requires ctx *8 for correct scaling
+                if(llamamodel->hparams.rope_freq_base_train==10000.0f && file_format_meta.n_tensors==435)
```

Comment:
what is `file_format_meta.n_tensors==435`

Reply:
It's to identify solar models because they need a different calculation of chi numbers.

Reply:
I had to come up with a way of identifying Solar models (and technically mistral v0.1, but i haven't found a decent way to do that) because the sliding window causes issues with calculating the optimal rope. 
The trained ctx, desired ctx have to be multiplied by 8 to match the fact that the sliding window is 1/8th of the 'full' context of the model. 
It's still the same formula, just scaled to the correct values.